### PR TITLE
feat(plugin-chart-pivot-table): add sort by result to data pane

### PR DESCRIPTION
**Pull request by kgabryje**
Adds "Sort by" metric's values to south data table (below the chart).

image

CC @junlincc

### CODE REVIEW

File: plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts [hunk 1]
```diff
@@ -16,21 +16,38 @@
...
+    const orderBy = ensureIsArray(timeseries_limit_metric);
+    if (
+      orderBy.length > 0 &&
+      ((typeof orderBy[0] === 'string' && !metrics?.includes(orderBy[0])) ||
+        !metrics?.find(
+          metric => (metric as AdhocMetric).label === (orderBy[0] as AdhocMetric).label,
+        ))
+    ) {
+      metrics?.push(orderBy[0]);
+    }
```
Comment (zhaoyongjie) [inline plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:45]: Why do we use orderby clause append to metrics?

File: plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts [hunk 2]
```diff
@@ -16,21 +16,38 @@
...
   const groupbySet = new Set([
     ...ensureIsArray<string>(groupbyColumns),
     ...ensureIsArray<string>(groupbyRows),
   ]);
   return buildQueryContext(formData, baseQueryObject => {
     const queryObject = normalizeOrderBy({ ...baseQueryObject, order_desc });
+    const { metrics } = queryObject;
+    const orderBy = ensureIsArray(timeseries_limit_metric);
+    if (
+      orderBy.length > 0 &&
```
Comment (villebro) [inline plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:38]: nit:
```suggestion
      orderBy.length &&
```

File: plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts (hunk 1 above)
Comment (kgabryje) [reply plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:45]: We want to append the values of metric that we sort by to the data table (below the chart)

Comment (villebro) [inline plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:45]: Can we use `getMetricLabel` here? https://github.com/apache-superset/superset-ui/blob/9b92a69d8eeb1c31548ce74a0538e70901c5bf26/packages/superset-ui-core/src/query/getMetricLabel.ts#L3-L14

**Review by villebro — COMMENTED** [review]
I recall there being discussion about this a while back (always adding orderbys to the select), but I'm not sure what the outcome was. But I'm ok with this until we decide one way or another.

File: plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts (hunk 1 above)
Comment (kgabryje) [reply plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:45]: Thank you, there are so many nice helper functions that I'm not aware of 😄

**Review by zhaoyongjie — APPROVED** [review]
The code, LGTM

**Pull request merged**